typing/ctype.ml

```diff
@@ -4533,24 +4533,39 @@ let same_constr env t1 t2 =
 let () =
   Env.same_constr := same_constr
 
-let maybe_pointer_type env typ =
+let rec get_type_repr env typ =
    match (repr typ).desc with
   | Tconstr(p, _args, _abbrev) ->
     begin try
       let type_decl = Env.find_type p env in
-      not type_decl.type_immediate
-    with Not_found -> true
+      type_decl.type_repr
+    with Not_found -> Repr_any
     (* This can happen due to e.g. missing -I options,
        causing some .cmi files to be unavailable.
        Maybe we should emit a warning. *)
     end
   | Tvariant row ->
       let row = Btype.row_repr row in
-      (* if all labels are devoid of arguments, not a pointer *)
-      not row.row_closed
-      || List.exists
-          (function
-            | _, (Rpresent (Some _) | Reither (false, _, _, _)) -> true
-            | _ -> false)
-          row.row_fields
-  | _ -> true
+      let is_direct (_, field) = match field with
+      | Rpresent (Some _) | Reither (false, _, _, _) -> false
+      | _ -> true
```

Comment:
I'm going to be super annoying, but I feel like this indentation style
- is not common in the codebase (you'll find it in parmatch, but not in many other places)
- is completely absent in this file.
- should be avoided (and I'm going to use the consistency fallacy to justify that)

You've used it in various places (it's probably some heritage of your coq background?) but I feel like the recommended style nowadays is something more like:
```ocaml
let foo x =
  match x with
  | Pattern ->
      code
  | Pattern' ->
      code
```

Do you mind changing that? :)